### DISCUSSION TITLE
Multiple Instance of Same Consumer

### DISCUSSION THREAD

**Discussion by oguzhantortop** [majusko/pulsar-java-spring-boot-starter#284] [Ideas]
Hi,
Is it possible to create multiple instances for the same consumer? This would be good for Shared Subscriptions like insert-only scenarios where the order of the messages is not essential.

**Comment by majusko** [comment]
Yes, it should be possible, but you will need to override `consumerName` and `subscriptionName` for the same value for each consumer. Otherwise, it will be autogenerated, and it will act as a different subscription - meaning it won't guarantee the delivery only to one subscriber in a group and instead, everyone will receive the message. Another option would be to use `Exclusive` subscription with unique subscription and consumer names. Depends on your use case. I would recommend playing with the tests in Example project: https://github.com/majusko/java-pulsar-example
